### ISSUE TITLE
Support DOM injections via mutation observers

### ISSUE THREAD

**Issue by dy** [guybedford/es-module-shims#101]
Would that be of any interest implementing importmaps as polyfill - reduced version of es-module-shims, enabling only importmaps without `type="importmap-shim"` and `type="module-shim"` attributes?
Intercepting all scripts on the page before execution is possible as
```js
<script id='import-maps-polyfill'>
// intercept all subsequent scripts before init
(new MutationObserver(rx=>rx.forEach(({target:s}) => {
  if (s.tagName !== 'SCRIPT') return

  s.textContent = s.textContent.replace(
  // find any static import
  /(from\s+|import\s+)['"]([\w\-]+)['"]/g,
  (unmodified, action, selector) => {
    // If we can find a mapped path...
    const mapped = deps[selector]
    return mapped ?
      // ..then update the import to use that mapped URL instead.
      `${action}/* ${selector} */ '${mapped}'` :
      unmodified;
  });

  s.handled=true
}))).observe(document, {childList:true,subtree:true})
</script>

```

**Comment by guybedford** [comment]
es-module-shims basically _is_ an import maps polyfill - that's the majority of what the code does in rewriting the imports.

I'm all for implementing mutation observers by default - a PR would be very welcome. See https://github.com/guybedford/es-module-shims#dynamic-import-map-updates for an implementation example.

**Comment by guybedford** [comment]
I've added support for this in https://github.com/guybedford/es-module-shims/pull/113.

**Comment by guybedford** [comment]
Released in 0.10.0.

**Comment by dy** [comment]
Nice, now it's fully transparent polyfill for importmaps, with no-API (almost) - awesome!
